File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        HashMap<String,Integer> completionHashMaps=new HashMap<>();

        for(String s : completion){
            if(completionHashMaps.containsKey(s))
                completionHashMaps.put(s,completionHashMaps.get(s)+1);
            else
                completionHashMaps.put(s,1);
        }
        for(String s : participant){
            if(!completionHashMaps.containsKey(s)||completionHashMaps.get(s)==0){
                answer=s;
                break;
            }           
            completionHashMaps.put(s,completionHashMaps.get(s)-1);
        }
        
        
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    total=dict()
    for i in completion:
        if i not in total:
            total[i]=1
        else:
            total[i]+=1
    
    for i in participant:
        if i not in total:
            return i
        else:
            total[i]-=1
            if total[i]<0:
                return i